segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -1010,6 +1013,34 @@ private List<Long> getLedgerIdsToDelete(LogMetadata oldMetadata, LogMetadata cur
 
     //endregion
 
+    /**
+     * Deletes all ledgers followed by/including given ledgerId
+     *
+     * @param startId     starting ledger id to delete
+     */
+
+    @VisibleForTesting
+    int deleteLedgersStartingWithId(long startId) throws DataLogInitializationException {
+        AtomicInteger count = new AtomicInteger(0);
+        LogMetadata metadata = loadMetadata();
+        val ledgersToDelete = metadata.getLedgers().stream()
+                .map(LedgerMetadata::getLedgerId)
+                .filter(ledgerId -> ledgerId >= startId )
+                .collect(Collectors.toList());
+
+        log.info("List of ledgers to be deleted are: " + ledgersToDelete);
```

Comment:
If you observe how logs are being printed, they have always the same pattern: 
```log.warn("{}: Unable to verify persisted metadata (path = '{}{}').", this.traceObjectId, this.zkClient.getNamespace(), this.logNodePath, ex);```
That is, all the logs print the `this.traceObjectId` followed by a `:`, and then the log message. This is useful for debugging, so we have a consistent format. Please, keep this in mind when writing log messages.